src/attr.ts

```diff
@@ -66,13 +66,23 @@ export function initializeAttrs(instance: HTMLElement, names?: Iterable<string>)
         }
       }
     }
+    descriptor.configurable = true
     Object.defineProperty(instance, key, descriptor)
     if (key in instance && !instance.hasAttribute(name)) {
       descriptor.set!.call(instance, value)
     }
   }
 }
 
+export function removeAttrs(instance: HTMLElement, names?: Iterable<string>): void {
+  if (!names) names = getAttrNames(Object.getPrototypeOf(instance))
+  for (const key of names) {
+    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+    // @ts-ignore
+    delete instance[key]
+  }
+}
```

Comment:
Are we concerned about removing the properties on tear down? Seems like they're not doing anything too dangerous by being left. The chief concern of this PR seems to be to set `configurable: true`

Reply:
> Seems like they're not doing anything too dangerous by being left. The chief concern of this PR seems to be to set `configurable: true`

Yeah the PR could be just `configurable: true` on the properties when we define them. That would mean that the properties are then defined on the element when it's disconnected from the DOM and would just be re-defined when the element connects again.

I decided to remove the elements since it mirrors the behavior when the element is connected and the behavior would be consistent. If the element isn't connected to the DOM, the `@attr` properties aren't defined.

> Are we concerned about removing the properties on tear down? 

I'm not particularly concerned here. These attributes should be the ones that we defined in the first place, if they've changed in the life-time of the element, it should be reflected in the DOM so no data should be lost.